File: Editorial/0164_Maximum_Gap/CSharp/1_Comparison_Sorting.cs
public class Solution {
    public int MaximumGap(int[] nums) {
        if (nums == null ||
            nums.Length < 2) // check if array is empty or small sized
            return 0;

        Array.Sort(nums); // sort the array

        int maxGap = 0;

        for (int i = 0; i < nums.Length - 1; i++)
            maxGap = Math.Max(nums[i + 1] - nums[i], maxGap);

        return maxGap;
    }
}

File: Editorial/0101-0200/0164_Maximum_Gap/Java/1_Comparison_Sorting.java
public class Solution {
    public int maximumGap(int[] nums) {
        if (
            nums == null || nums.length < 2
        ) return 0; // check if array is empty or small sized

        Arrays.sort(nums); // sort the array

        int maxGap = 0;

        for (int i = 0; i < nums.length - 1; i++) maxGap = Math.max(
            nums[i + 1] - nums[i],
            maxGap
        );

        return maxGap;
    }
}
